sdk/include/sdk/AsyncResult.h

```diff
@@ -163,6 +163,17 @@ template <typename TResultType> class AsyncResult {
      */
     [[nodiscard]] bool isInAwaitingState() const { return m_awaiting; }
 
+    template <typename TType>
```

Comment:
Doxygen missing.

Reply:
@doosuu: Could you add some - I'm not exactly aware what this function is doing?

Reply:
@BjoernAtBosch added commit with doxygen documentation added.